### ISSUE TITLE
Dragging unity from 1 screen to another with different DPI causes training to crash

### ISSUE THREAD

**Issue by devedse** [Unity-Technologies/ml-agents#2748]
**Describe the bug**
When I have 2 monitors connected to my laptop and drag unity from my 200% DPI screen to my 100% DPI screen Unity takes a while to resize and then shows an exception. The agents also stop moving after this.

**To Reproduce**
1. Start training on one of the scenes using for example:
mlagents-learn config/trainer_config.yaml --run-id=hide_8 --train
2. Once training, drag Unity to your other DPI screen
3. Watch the Errors

**Console logs / stack traces**

442+ times:
```
NullReferenceException: Object reference not set to an instance of an object
MLAgents.Academy.EnvironmentStep () (at Assets/ML-Agents/Scripts/Academy.cs:522)
MLAgents.Academy.FixedUpdate () (at Assets/ML-Agents/Scripts/Academy.cs:583)
```

and
1 time:

```
NullReferenceException: Object reference not set to an instance of an object
MLAgents.Academy.OnDestroy () (at Assets/ML-Agents/Scripts/Academy.cs:596)
```

**Screenshots**
![image](https://user-images.githubusercontent.com/2350015/67032439-e169f100-f113-11e9-8def-9b28e76f2479.png)

**Environment (please complete the following information):**
- OS + version: Windows 10
- _ML-Agents version_: Latest develop branch
- _Environment_: Reproducable in all samples
- Unity version: 2019.1.14f1

**Comment by surfnerd** [comment]
Hey @devedse,
Thanks for the report.  We will take look at this.

**Comment by surfnerd** [comment]
Hey @devedse, 
There is a pull request open #2782 that potentially fixes this issue.  When you get a chance to try it out please let us know.

**Comment by surfnerd** [comment]
Hey @devedse,
I spoke with our engine team and it seems like this might be a bug in the editor.  It's causing a script reload when you drag between windows with different DPIs at the moment, which seems like a bit of overkill.  This pull request will indeed fix the errors from showing up in the console, but your entire scene is being reloaded when you drag between those different DPI screens.  In other words, if you are in play mode on windows, don't drag the editor between screens if you don't want to reload your entire scene.😢 

I am filing a bug against the editor internally and will post back here when it is fixed.

**Comment by surfnerd** [comment]
@devedse,
There has been an issue filed against Unity with the ID 1193597.  I am going to close this for now, and I'll update it when the fix goes into a Unity release.  Until then, feel free to open another issue if you run into trouble.
Cheers,
Chris and the ML-Agents Team

**Comment by surfnerd** [comment]
Hi, the fix for this will be in the 2019.3 release of unity.  It looks like it may be in 2019.3.0.b10 release if you are following the beta releases.

**Comment by surfnerd** [comment]
You can find it in the release notes here https://unity3d.com/unity/beta/2019.3#downloads
```Editor: Prevent reloading scripts in playmode when a DPI change occurs (1193597) ```
